src/v/security/acl_store.h

```diff
@@ -80,6 +80,11 @@ class acl_matches {
 
     using entry_set_ref = acl_entry_set_match;
 
+    struct acl_match {
+        std::reference_wrapper<const resource_pattern> resource_pattern;
+        acl_entry_set::const_reference acl;
+    };
+
```

Comment:
I questioned the importance of this structure compared to `acl_entry_set_match` on the initial review since the two structures have similar contents. The next commit, however, reveals why this is here. Is it possible to use a single structure? OK if not.

Reply:
Completely understand this question.  So to answer this, we have to step through how an ACL match is found.  ACLs are tied to resource patterns.  The first step is finding which `resource_pattern` matches the requested resource (which is the templated variable of type `T`).  That `resource_pattern` is associated with a set (`acl_entry_set`) of ACLs.  Originally, the code just returned the `acl_entry_set`.  But for auditing, I also want to know which `resource_pattern` matched.  So that's why I have `acl_entry_set_match`.  Once that's returned, we search through that `acl_entry_set` to get the final matched `acl` which is then bundled with the matched `resource_pattern` to return `acl_match`.

Reply:
Beautiful explanation